lib/promoted/ruby/client/request_builder.rb

```diff
@@ -115,49 +96,27 @@ def default_request_values
           @default_request_values
         end
 
-        def only_log
-          @only_log
-        end
-
-        def full_insertion
-          @full_insertion
-        end
-
-        def user_info
-          @user_info
-        end
-
-        def timing
-          @timing
-        end
-
-        def request_id
-          @request_id
-        end
-
-        def log_request_params
-          {
+        def log_request_params(include_insertions: true, include_request: true)
+          params = {
             user_info: user_info,
             timing: timing,
-            request: [request],
-            insertion: compact_insertions
+            cohort_membership: @experiment
           }
+          params[:request] = [request] if include_request
+          params[:insertion] = compact_metrics_insertions if include_insertions
+          
+          params.clean!
         end
 
-        def request_params include_insertion: true
-          @request_params = {
-            user_info: user_info,
-            timing: timing,
-            request_id: request_id,
-            view_id: view_id,
-            session_id: session_id,
-            insertion: compact_insertions
-          }
-          @request_params.merge!({insertion: compact_insertions}) if include_insertion
-          @request_params
+        def compact_delivery_insertions
```

Comment:
might make sense to rename `compact_metrics_insertions` to be structured more like this.  not needed in this pr.

Reply:
Previously I had _almost_ put a comment on compact_metrics_insertions to the effect of "reconsider, this looks overly complicated". You inspired me to go ahead and add that.